### DISCUSSION TITLE
[QA] 想咨询个Circle组件问题

### DISCUSSION THREAD

**Discussion by sphenginx** [youzan/vant#11713] [Q&A] [answered]
关于Circle组件，有个  stroke-width 属性，[文档介绍只有  “进度条宽度”](https://vant-ui.github.io/vant/#/zh-CN/circle)， 并未标明单位…… 

想问下 这个属性的单位是啥？

比如UI设定了宽度是 10px， stroke-width 要传多少？我试了10宽度很窄，传了100 还感觉比UI的要窄一些…… 

下图 左边 是我设置了 stroke-width = 100 的值， 右边是UI 设定了 border-width 是 10px

![img_v2_a81c263a-5d8e-4d6c-9900-9618d5a9ec5g](https://user-images.githubusercontent.com/1829395/228205110-f1a9c261-5fdb-45b1-9623-45f6fad2e826.jpg)

vant版本相关如下：
```json
"vant": {
      "version": "4.0.10",
      "resolved": "https://registry.npmmirror.com/vant/-/vant-4.0.10.tgz",
      "integrity": "sha512-kpxzjgjJdZePqbcG0q+CM4v5lGFlPxQMb2irXsknUzHFe8XZgk3cQ/gukpn9flpMuNi6hPcwoQaAnyzAKlCYpg==",
      "requires": {
        "@vant/popperjs": "^1.3.0",
        "@vant/use": "^1.4.4"
      }
    },
```

**Comment by wChenonly** [comment]
单位是px,这个设置的是svg的stroke-width

**Reply by sphenginx** [reply]
不对吧，我设置了 100，但是宽度 咋还不如 UI设置的 10px 宽……

**Comment by chenjiahan** [comment, marked as answer]
`stroke-width` 指的是 SVG 中 `path` 的宽度，默认值为 `40`。

`stroke-width` 的单位不是 `px`，如果你想知道 `stroke-width` 与 `px` 的换算关系，可以通过如下公式计算：

```js
// SVG 的 viewBox 大小
const viewBox = 1000 + strokeWidth;
// Circle 组件的宽度，默认为 100px
const circleWidth = 100;
// 最终渲染出来的进度条宽度（px）
const pxWidth = (strokeWidth * circleWidth) / viewBox;
```

**Reply by sphenginx** [reply]
建议这个说明 可以 输出到 vant的文档里面……

**Reply by chenjiahan** [reply]
上周已经在文档里添加了相关内容

**Reply by sphenginx** [reply]
嗯，刚看了下，在宽度定制那里 加了 上面回答的说明。
